src/github.com/couchbase/sync_gateway/rest/api_test.go

```diff
@@ -246,6 +269,55 @@ func TestFunkyDocIDs(t *testing.T) {
 	assertStatus(t, response, 200)
 }
 
+func TestCORSOrigin(t *testing.T) {
+	var rt restTester
+	reqHeaders := map[string]string{
+		"Origin": "http://example.com",
+	}
+	response := rt.sendRequestWithHeaders("GET", "/db/", "", reqHeaders)
+	assert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "http://example.com")
+
+	// now test a non-listed origin
+	// b/c * is in config we get *
+	reqHeaders = map[string]string{
+		"Origin": "http://hack0r.com",
+	}
+	response = rt.sendRequestWithHeaders("GET", "/db/", "", reqHeaders)
+	assert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "*")
+
+	// now test another origin in config
+	reqHeaders = map[string]string{
+		"Origin": "http://staging.example.com",
+	}
+	response = rt.sendRequestWithHeaders("GET", "/db/", "", reqHeaders)
+	assert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "http://staging.example.com")
+
+	// todo test with a config without * should reject non-matches
```

Comment:
Should probably get this test added, to make sure this doesn't get broken at some point in future.